### PR TITLE
planner: update the doc about Plan Cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.1 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-hints.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
 > **警告：**
 >
 > 强烈建议不要利用此 Hint 修改没有明确支持的变量，这可能会引发不可预知的行为。
+> `SET_VAR` 请尽量写在最外层而不要写在子查询中，否则可能会不生效。
```
Comment (time-and-fate) [inline optimizer-hints.md:86]: 这个只是 SET_VAR hint 的文档放错位置了吧，它不是个“查询块范围生效的 Hint”，而是“查询范围生效的 Hint”。
我们在“查询范围生效的 Hint”一节里面是已经有文档说明这个问题的：

> 这类 Hint 只能跟在语句中第一个 SELECT、UPDATE 或 DELETE 关键字的后面

File: sql-prepared-plan-cache.md [hunk 2]
```diff
@@ -180,6 +182,23 @@ mysql> SHOW WARNINGS;
 1 row in set (0.00 sec)
 ```
 
+### 通过 Statement Summary 诊断
```
Comment (time-and-fate) [inline sql-prepared-plan-cache.md:185]:
```suggestion
### 通过 Statements Summary 诊断
```

File: sql-prepared-plan-cache.md [hunk 3]
```diff
@@ -180,6 +182,23 @@ mysql> SHOW WARNINGS;
...
+
+在 `Statement Summary` 表中包含有 `plan_cache_unqualified` 和 `last_plan_cache_unqualified_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
```
Comment (time-and-fate) [inline sql-prepared-plan-cache.md:187]:
```suggestion
在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 `last_plan_cache_unqualified_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
```

Comment (time-and-fate) [reply sql-prepared-plan-cache.md:187]: 以及，是不是需要在 `statement-summary-tables.md` 页面最后那一部分加上对应的说明。

File: sql-prepared-plan-cache.md [hunk 4]
```diff
@@ -148,6 +148,8 @@ MySQL [test]> select @@last_plan_from_cache;
 
 ## 诊断 Prepared Plan Cache
 
+### 通过 Explain 诊断
```
Comment (time-and-fate) [inline sql-prepared-plan-cache.md:151]: 通过 SHOW WARNINGS 吧？这一节也没 EXPLAIN 呀。🤣

File: optimizer-hints.md (hunk 1 above)
Comment (songrijie) [reply optimizer-hints.md:86]: +1

File: sql-prepared-plan-cache.md (hunk 3 above)
Comment (songrijie) [reply sql-prepared-plan-cache.md:187]: `last_plan_cache_unqualified_reason` ，建议 `plan_cache_unqualified_last_reason`

**Comment by songrijie** [comment]
还需要向 https://docs.pingcap.com/zh/tidb/stable/statement-summary-tables 增加字段说明

File: sql-prepared-plan-cache.md (hunk 4 above)
Comment (qw4990) [inline sql-prepared-plan-cache.md:151]:
```suggestion
### 通过 Show Warnings 诊断 
```

File: optimizer-hints.md (hunk 1 above)
Comment (qw4990) [reply optimizer-hints.md]: 已修改，把 set_var hint 的说明挪到 “查询范围内生效的 Hint” 内去了，不过还是补了这个 notice，免得用户没注意，直接吧 set_var 写到子查询内去导致不生效；

File: sql-prepared-plan-cache.md (hunk 4 above)
Comment (qw4990) [reply sql-prepared-plan-cache.md]: 已修正

File: statement-summary-tables.md [hunk 5]
```diff
@@ -279,6 +279,8 @@ SQL 的基础信息：
 - `BINARY_PLAN`：以二进制格式编码后的原执行计划，存在多条语句时，只取其中一条语句的执行计划。用 `select tidb_decode_binary_plan('xxx...')` SQL 语句可以解析出具体的执行计划。
 - `PLAN_CACHE_HITS`：这类 SQL 语句命中 plan cache 的总次数
 - `PLAN_IN_CACHE`：这类 SQL 语句的上次执行是否命中了 plan cache
+- `PLAN_CACHE_UNQUALIFIED`：这类 SQL 没法命中 Plan Cache 的次数
```
Comment (time-and-fate) [inline statement-summary-tables.md:282]: 这个和总执行次数减去 `PLAN_CACHE_HITS` 的区别是什么呢？

**Review by songrijie — APPROVED** [review]

File: statement-summary-tables.md (hunk 5 above)
Comment (qw4990) [reply statement-summary-tables.md:282]: 一是单独用个字段清晰点，用户不太会仔细思考去做加减法，二是语义上有点不同，如果是 cache 满了导致无法命中，而不是查询本身收到限制导致无法命中，不会被记录在里面； @time-and-fate

Comment (time-and-fate) [reply statement-summary-tables.md:282]: 好的

**Review by time-and-fate — APPROVED** [review]

File: optimizer-hints.md [hunk 6]
```diff
@@ -818,6 +787,39 @@ SELECT /*+ READ_CONSISTENT_REPLICA() */ * FROM t;
...
 ```
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` 允许在语句执行期间以 Hint 形式临时修改会话级系统变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。支持通过 `SET_VAR(VAR_NAME=VAR_VALUE)` Hint 修改的系统变量请查看[系统变量](/system-variables.md)。
+
+> **警告：**
+>
+> 强烈建议不要利用此 Hint 修改没有明确支持的变量，这可能会引发不可预知的行为。
+> 请注意不要把 `SET_VAR` 写在子查询中，否则可能会不生效。
```
Comment (hfxsd) [inline optimizer-hints.md:797]:
```suggestion
> 请注意不要把 `SET_VAR` 写在子查询中，否则可能会不生效。详情请参考 [`SET_VAR` 写在子查询中不生效](#set_var-写在子查询中不生效)。
```

File: optimizer-hints.md [hunk 7]
```diff
@@ -1041,3 +1043,25 @@ set tidb_opt_enable_hash_join=off;
 EXPLAIN SELECT /*+ NO_MERGE_JOIN(t1) */ * FROM t1, t2 WHERE t1.a=t2.a;
 ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
 ```
+
+### `SET_VAR` 写在子查询中不生效
+
+`SET_VAR` 用来设置当前语句的系统变量，需要尽量写在最外层，如果写在子查询中，由于子查询会被特殊处理，可能导致 `SET_VAR` 无法生效：
```
Comment (hfxsd) [inline optimizer-hints.md:1049]:
```suggestion
`SET_VAR` 用来设置当前语句的系统变量，需要尽量写在最外层。如果写在子查询中，由于子查询会被特殊处理，可能导致 `SET_VAR` 无法生效。

下面示例把 `SET_VAR` 写在了子查询中，所以没有生效。
```

File: optimizer-hints.md [hunk 8]
```diff
@@ -1041,3 +1043,25 @@ set tidb_opt_enable_hash_join=off;
...
+
+```sql
+mysql> SELECT @@MAX_EXECUTION_TIME, a FROM (SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=123) */ 1 as a) t;
++----------------------+---+
+| @@MAX_EXECUTION_TIME | a |
++----------------------+---+
+|                    0 | 1 |
++----------------------+---+
+1 row in set (0.00 sec)
+
```
Comment (hfxsd) [inline optimizer-hints.md:1059]:
```suggestion
```

下面示例把 `SET_VAR` 写在了最外层，所以可以生效。

```
```

File: sql-prepared-plan-cache.md [hunk 9]
```diff
@@ -148,6 +148,8 @@ MySQL [test]> select @@last_plan_from_cache;
 
 ## 诊断 Prepared Plan Cache
 
+### 通过 Show Warnings 诊断
```
Comment (hfxsd) [inline sql-prepared-plan-cache.md:151]:
```suggestion
### 通过 `SHOW WARNINGS` 诊断 
```

File: sql-prepared-plan-cache.md [hunk 10]
```diff
@@ -180,6 +182,23 @@ mysql> SHOW WARNINGS;
 1 row in set (0.00 sec)
 ```
 
+### 通过 Statements Summary 诊断
+
+在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 `last_plan_cache_unqualified_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
+
+```sql
+mysql> select digest_text, plan_cache_unqualified, last_plan_cache_unqualified_reason from information_schema.statements_summary where plan_cache_unqualified > 0 order by plan_cache_unqualified desc limit 10;
```
Comment (hfxsd) [inline sql-prepared-plan-cache.md:190]:
```suggestion
mysql> SELECT digest_text, plan_cache_unqualified, last_plan_cache_unqualified_reason FROM information_schema.statements_summary WHERE plan_cache_unqualified > 0 ORDER BY plan_cache_unqualified DESC
LIMIT 10;
```

File: statement-summary-tables.md (hunk 5 above)
Comment (hfxsd) [inline statement-summary-tables.md:282]:
```suggestion
- `PLAN_CACHE_UNQUALIFIED`：这类 SQL 没法命中 plan cache 的次数
```

File: statement-summary-tables.md [hunk 11]
```diff
@@ -279,6 +279,8 @@ SQL 的基础信息：
...
+- `LAST_PLAN_CACHE_UNQUALIFIED_REASON`：这类 SQL 没法命中 Plan Cache 的最后一次原因
```
Comment (hfxsd) [inline statement-summary-tables.md:283]:
```suggestion
- `LAST_PLAN_CACHE_UNQUALIFIED_REASON`：这类 SQL 最后一次没法命中 plan cache 的原因
```

File: optimizer-hints.md [hunk 12]
```diff
@@ -1041,3 +1043,31 @@ set tidb_opt_enable_hash_join=off;
...
+mysql> SELECT @@MAX_EXECUTION_TIME, a FROM (SELECT /*+ SET_VAR(MAX_EXECUTION_TIME=123) */ 1 as a) t;
++----------------------+---+
+| @@MAX_EXECUTION_TIME | a |
++----------------------+---+
+|                    0 | 1 |
++----------------------+---+
+1 row in set (0.00 sec)
+```
+
+下面示例把 SET_VAR 写在了最外层，所以可以生效。
```
Comment (hfxsd) [inline optimizer-hints.md:1063]:
```suggestion
下面示例没有把 SET_VAR 写在字查询中，所以可以生效。
```

File: optimizer-hints.md [hunk 13]
```diff
@@ -1041,3 +1043,31 @@ set tidb_opt_enable_hash_join=off;
 EXPLAIN SELECT /*+ NO_MERGE_JOIN(t1) */ * FROM t1, t2 WHERE t1.a=t2.a;
 ERROR 1815 (HY000): Internal : Can't find a proper physical plan for this query
 ```
+
+### `SET_VAR` 写在子查询中不生效
+
+`SET_VAR` 用来设置当前语句的系统变量，不要写在字查询中。如果写在子查询中，由于子查询会被特殊处理，可能导致 `SET_VAR` 无法生效。
```
Comment (hfxsd) [inline optimizer-hints.md:1049]:
```suggestion
`SET_VAR` 用来设置当前语句的系统变量，不要写在子查询中。如果写在子查询中，由于子查询会被特殊处理，可能导致 `SET_VAR` 无法生效。
```

File: statement-summary-tables.md [hunk 14]
```diff
@@ -279,6 +279,8 @@ SQL 的基础信息：
 - `BINARY_PLAN`：以二进制格式编码后的原执行计划，存在多条语句时，只取其中一条语句的执行计划。用 `select tidb_decode_binary_plan('xxx...')` SQL 语句可以解析出具体的执行计划。
 - `PLAN_CACHE_HITS`：这类 SQL 语句命中 plan cache 的总次数
 - `PLAN_IN_CACHE`：这类 SQL 语句的上次执行是否命中了 plan cache
+- `PLAN_CACHE_UNQUALIFIED`：这类 SQL 没法命中 plan cache 的次数
+- `LAST_PLAN_CACHE_UNQUALIFIED_REASON`：这类 SQL 最后一次没法命中 plan cache 的原因
```
Comment (hfxsd) [inline statement-summary-tables.md:283]:
```suggestion
- `PLAN_CACHE_UNQUALIFIED`：这类 SQL 语句没有命中 plan cache 的次数
- `LAST_PLAN_CACHE_UNQUALIFIED_REASON`：这类 SQL 语句最后一次没有命中 plan cache 的原因
```

File: sql-prepared-plan-cache.md [hunk 15]
```diff
@@ -180,6 +182,24 @@ mysql> SHOW WARNINGS;
 1 row in set (0.00 sec)
 ```
 
+### 通过 Statements Summary 诊断
```
Comment (hfxsd) [inline sql-prepared-plan-cache.md:185]:
```suggestion
### 通过 `Statements Summary` 诊断
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by hfxsd — APPROVED** [review]

File: optimizer-hints.md [hunk 16]
```diff
@@ -818,6 +787,39 @@ SELECT /*+ READ_CONSISTENT_REPLICA() */ * FROM t;
...
 ```
 
+### SET_VAR(VAR_NAME=VAR_VALUE)
+
+`SET_VAR(VAR_NAME=VAR_VALUE)` 允许在语句执行期间以 Hint 形式临时修改会话级系统变量的值。当语句执行完成后，系统变量将在当前会话中自动恢复为原始值。通过这个 Hint 可以修改一部分与优化器、执行器相关的系统变量行为。支持通过 `SET_VAR(VAR_NAME=VAR_VALUE)` Hint 修改的系统变量请查看[系统变量](/system-variables.md)。
+
+> **警告：**
+>
+> 强烈建议不要利用此 Hint 修改没有明确支持的变量，这可能会引发不可预知的行为。
+> 请注意不要把 `SET_VAR` 写在子查询中，否则可能会不生效。详情请参考 [`SET_VAR` 写在子查询中不生效](#set_var-写在子查询中不生效)。
```
Comment (lilin90) [inline optimizer-hints.md:797]:
```suggestion
> - 强烈建议不要利用此 Hint 修改没有明确支持的变量，这可能会引发不可预知的行为。
> - 注意不要把 `SET_VAR` 写在子查询中，否则可能会不生效。详情请参考 [`SET_VAR` 写在子查询中不生效](#set_var-写在子查询中不生效)。
```

File: sql-prepared-plan-cache.md [hunk 17]
```diff
@@ -180,6 +182,24 @@ mysql> SHOW WARNINGS;
...
 
+### 通过 `Statements Summary` 诊断
+
+在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 `last_plan_cache_unqualified_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
+
+```sql
+mysql> SELECT digest_text, plan_cache_unqualified, last_plan_cache_unqualified_reason FROM information_schema.statements_summary WHERE plan_cache_unqualified > 0 ORDER BY plan_cache_unqualified DESC
+LIMIT 10;
++---------------------------------+------------------------+----------------------------------------+
+| digest_text                     | plan_cache_unqualified | last_plan_cache_unqualified_reason     |
```
Comment (hfxsd) [inline sql-prepared-plan-cache.md:193]:
```suggestion
在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 ` plan_cache_unqualified_last_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：

```sql
mysql> SELECT digest_text, plan_cache_unqualified,  plan_cache_unqualified_last_reason FROM information_schema.statements_summary WHERE plan_cache_unqualified > 0 ORDER BY plan_cache_unqualified DESC
LIMIT 10;
+---------------------------------+------------------------+----------------------------------------+
| digest_text                     | plan_cache_unqualified | plan_cache_unqualified_last_reason     |
```

File: optimizer-hints.md [hunk 18]
```diff
@@ -76,37 +76,6 @@ SELECT /*+ QB_NAME(QB1) */ * FROM (SELECT * FROM t) t1, (SELECT * FROM t) t2;
 >
 > 上述例子中，如果指定的 `QB_NAME` 为 `sel_2`，并且不给原本 `sel_2` 对应的第二个查询块指定新的 `QB_NAME`，则第二个查询块的默认名字 `sel_2` 会失效。
```
Comment (hfxsd) [inline optimizer-hints.md:78]:
```suggestion
```

File: sql-prepared-plan-cache.md [hunk 19]
```diff
@@ -180,6 +182,24 @@ mysql> SHOW WARNINGS;
 1 row in set (0.00 sec)
 ```
 
+### 通过 `Statements Summary` 诊断
+
+在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 ` plan_cache_unqualified_last_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
```
Comment (hfxsd) [inline sql-prepared-plan-cache.md:187]:
```suggestion
在 `Statements Summary` 表中包含有 `plan_cache_unqualified` 和 `plan_cache_unqualified_last_reason` 两个字段，分别表示对应查询无法使用 Plan Cache 的次数和原因，可以通过这两个字段来进行诊断：
```

File: optimizer-hints.md (hunk 18 above)
Comment (hfxsd) [inline optimizer-hints.md:78]:
```suggestion
```

**Pull request merged**